### ISSUE TITLE
Typing `xml"` will crash the REPL

### ISSUE THREAD

**Issue by Glavo** [lampepfl/dotty#6676]
When typing `xml"`, the REPL crashed: 
```
glavo@glavo:~$ dotr
Starting dotty REPL...
scala> xmlException in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 4 out of bounds for length 4                                                                      
	at dotty.tools.dotc.parsing.Parsers$Parser.interpolatedString(Parsers.scala:802)
	at dotty.tools.dotc.parsing.Parsers$Parser.literal(Parsers.scala:790)
	at dotty.tools.dotc.parsing.Parsers$Parser.simpleExpr(Parsers.scala:1636)
	at dotty.tools.dotc.parsing.Parsers$Parser.$init$$$anonfun$6(Parsers.scala:1572)
	at dotty.tools.dotc.parsing.Parsers$Parser.postfixExpr(Parsers.scala:1559)
	at dotty.tools.dotc.parsing.Parsers$Parser.expr1(Parsers.scala:1403)
	at dotty.tools.dotc.parsing.Parsers$Parser.expr(Parsers.scala:1306)
	at dotty.tools.dotc.parsing.Parsers$Parser.blockStatSeq$$anonfun$1(Parsers.scala:3112)
	at dotty.tools.dotc.parsing.Parsers$Parser.checkNoEscapingPlaceholders(Parsers.scala:444)
	at dotty.tools.dotc.parsing.Parsers$Parser.blockStatSeq(Parsers.scala:3133)
	at dotty.tools.dotc.printing.SyntaxHighlighting$.highlight(SyntaxHighlighting.scala:121)
	at dotty.tools.repl.JLineTerminal$Highlighter.highlight(JLineTerminal.scala:76)
	at org.jline.reader.impl.LineReaderImpl.getHighlightedBuffer(LineReaderImpl.java:3633)
	at org.jline.reader.impl.LineReaderImpl.getDisplayedBufferWithPrompts(LineReaderImpl.java:3615)
	at org.jline.reader.impl.LineReaderImpl.redisplay(LineReaderImpl.java:3554)
	at org.jline.reader.impl.LineReaderImpl.doCleanup(LineReaderImpl.java:2340)
	at org.jline.reader.impl.LineReaderImpl.cleanup(LineReaderImpl.java:2332)
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:626)
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:404)
	at dotty.tools.repl.JLineTerminal.readLine(JLineTerminal.scala:68)
	at dotty.tools.repl.ReplDriver.readLine$1(ReplDriver.scala:115)
	at dotty.tools.repl.ReplDriver.loop$1(ReplDriver.scala:125)
	at dotty.tools.repl.ReplDriver.runUntilQuit$$anonfun$1(ReplDriver.scala:130)
	at dotty.tools.repl.ReplDriver.withRedirectedOutput$$anonfun$2$$anonfun$1(ReplDriver.scala:143)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withErr(Console.scala:196)
	at dotty.tools.repl.ReplDriver.withRedirectedOutput$$anonfun$1(ReplDriver.scala:143)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withOut(Console.scala:167)
	at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:143)
	at dotty.tools.repl.ReplDriver.runUntilQuit(ReplDriver.scala:130)
	at dotty.tools.repl.Main$.main(Main.scala:6)
	at dotty.tools.repl.Main.main(Main.scala)
```

**Comment by SethTisue** [comment]
I'm not able to reproduce this in either 0.16.0-RC3  or on master

**Comment by OlivierBlanvillain** [comment]
Reproduced, but it seams to be unrelated to `xml`. A normal `s` interpolator also breaks the repl:

(typing `dotr` then `s"`)

```scala
~/workspace/dotty-master (master) $ dotr                                       
Starting dotty REPL...
scala> sException in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
[...]         
```

**Comment by SethTisue** [comment]
> typing `dotr`

in what context? from the unix shell? in sbt?

I'm typing `dotr` from inside sbt, and I also tried running `/usr/local/bin/dotr` using homebrew-installed 0.16.0-RC3

I also tried going the `sbt new lampepfl/dotty.g8` and then running `sbt console` and wasn't able to reproduce it either with 0.15.0-RC1 or 0.16.0-RC3 (changing the version in `build.sbt`)

I only belabor all this because it seems like a likely clue to the nature of the problem.

I'm on Mac OS X 10.14.5, Java 1.8.0_202
